cli/azd/pkg/project/service_target_aks.go

```diff
@@ -0,0 +1,434 @@
+	}
+
+	if err := kubeConfigManager.MergeConfigs(ctx, "config", "config", clusterName); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed merging kube configs: %w", err)
+	}
+
+	if _, err := t.kubectl.ConfigUseContext(ctx, clusterName, nil); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed using kube context '%s', %w", clusterName, err)
+	}
+
+	kubeFlags := kubectl.KubeCliFlags{
+		Namespace: namespace,
+		DryRun:    "client",
+		Output:    "yaml",
+	}
+
+	progress <- "Creating k8s namespace"
+	namespaceResult, err := t.kubectl.CreateNamespace(
+		ctx,
+		namespace,
+		&kubectl.KubeCliFlags{DryRun: "client", Output: "yaml"},
+	)
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed creating kube namespace: %w", err)
+	}
+
+	_, err = t.kubectl.ApplyPipe(ctx, namespaceResult.Stdout, nil)
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed applying kube namespace: %w", err)
+	}
+
+	progress <- "Creating k8s secrets"
+	secretResult, err := t.kubectl.CreateSecretGenericFromLiterals(ctx, "azd", t.env.Environ(), &kubeFlags)
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed setting kube secrets: %w", err)
+	}
+
+	_, err = t.kubectl.ApplyPipe(ctx, secretResult.Stdout, nil)
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed applying kube secrets: %w", err)
+	}
+
+	// Login to container registry.
+	loginServer, has := t.env.Values[environment.ContainerRegistryEndpointEnvVarName]
+	if !has {
+		return ServiceDeploymentResult{}, fmt.Errorf(
+			"could not determine container registry endpoint, ensure %s is set as an output of your infrastructure",
+			environment.ContainerRegistryEndpointEnvVarName,
+		)
+	}
+
+	log.Printf("logging into registry %s\n", loginServer)
+
+	progress <- "Logging into container registry"
+	if err := t.az.LoginAcr(ctx, t.docker, t.env.GetSubscriptionId(), loginServer); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("logging into registry '%s': %w", loginServer, err)
+	}
+
+	imageTag, err := t.generateImageTag()
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("generating image tag: %w", err)
+	}
+
+	fullTag := fmt.Sprintf(
+		"%s/%s",
+		loginServer,
+		imageTag,
+	)
+
+	// Tag image.
+	log.Printf("tagging image %s as %s", path, fullTag)
+	progress <- "Tagging image"
+	if err := t.docker.Tag(ctx, t.config.Path(), path, fullTag); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("tagging image: %w", err)
+	}
+
+	log.Printf("pushing %s to registry", fullTag)
+
+	// Push image.
+	progress <- "Pushing container image"
+	if err := t.docker.Push(ctx, t.config.Path(), fullTag); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("pushing image: %w", err)
+	}
+
+	// Save the name of the image we pushed into the environment with a well known key.
+	t.env.SetServiceProperty(t.config.Name, "IMAGE_NAME", fullTag)
```

Comment:
The `SetServiceProperty()` func handles this by using the azd service name in the env var.  Example for an `api` app, it will construct an env var like `SERVICE_API_IMAGE_NAME`.  This is consistent on how container apps work today as well.